scripts/upgrade-test.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+# Runs a test to see what happens if we upgrade consensus
+
+set -euxo pipefail
+export RUST_LOG="${RUST_LOG:-info}"
+
+source ./scripts/setup-tests.sh
+
+server4=$(tail -4 $FM_PID_FILE | head -1)
+server3=$(tail -3 $FM_PID_FILE | head -1)
+server2=$(tail -2 $FM_PID_FILE | head -1)
+server1=$(tail -1 $FM_PID_FILE | head -1)
+
+
+function wait_server_shutdown() {
+  echo "Waiting for $1 to shutdown..."
+  tail --pid=$1 -f /dev/null
+  echo "Server $1 has shutdown."
+}
+
+await_fedimint_block_sync
+
+# test a consensus upgrade
+FM_PASSWORD="pass0" $FM_MINT_CLIENT signal-upgrade $FM_CFG_DIR/server-0/private.salt 0
+FM_PASSWORD="pass1" $FM_MINT_CLIENT signal-upgrade $FM_CFG_DIR/server-1/private.salt 1
```

Comment:
I think the best security option is to put the salt in the admin UI.  Failing that, it would live on fedimintd and not be passed.

Reply:
Make the `salt == hash(federationid || peer_id)` or something? AFAIK it doesn't have to be secret, just different to prevent rainbow tables on some other forms of mass-breaking of keys, no?

Reply:
It doesn't have to be secret, although if you make it secret it does help users with low-entropy passwords.

Although in all likelihood it's probably more of a foot-gun to depend on users to back it up properly so maybe we should just derive it.